tests/unit_tests/test_tokenizer.py

```diff
@@ -0,0 +1,410 @@
+            "\n\t\r",  # Whitespace characters
+            "a" * 1000,  # Very long repeated character
+        ]
+
+        for text in edge_cases:
+            # Handle different tokenizer types for edge cases too
+            if hasattr(our_tokenizer, "tokenizer"):
+                our_tokens = our_tokenizer.encode(text)
+            else:
+                our_encoded = our_tokenizer.encode(text)
+                our_tokens = (
+                    our_encoded.ids if hasattr(our_encoded, "ids") else our_encoded
+                )
+
+            reference_tokens = encode_text(reference_tokenizer, text)
+
+            self.assertEqual(
+                our_tokens,
+                reference_tokens,
+                f"Edge case tokens should match for text '{text[:50]}...' in {test_repo_id} ({tokenizer_type})",
+            )
+
+    @parametrize(
+        "test_repo_id",
+        [
+            "meta-llama/Meta-Llama-3.1-8B",
+            "deepseek-ai/DeepSeek-V3",
+            # "black-forest-labs/FLUX.1-dev", TODO: load the actual tokenizer
+            "Qwen/Qwen2-7B",
+        ],
+    )
+    def test_download_and_build_tokenizer(self, test_repo_id):
+        """
+        Test downloading tokenizer files and loading them, comparing with official APIs.
+
+        This test:
+        1. Downloads tokenizer files using download_hf_tokenizer_files
+        2. Loads tokenizer using our load_tokenizer function
+        3. Compares behavior with official Tokenizer library
+        4. Compares with transformers AutoTokenizer (if available)
+        """
+        # Step 1: Download tokenizer files
+        try:
+            download_hf_tokenizer_files(
+                repo_id=test_repo_id,
+                local_dir=self.temp_dir,
+            )
+        except HTTPError as e:
+            if test_repo_id == "meta-llama/Meta-Llama-3.1-8B":
+                self.skipTest(
+                    f"Could not download tokenizer files for Meta-Llama-3.1-8B: {e}"
+                )
+            else:
+                raise e
+
+        # Step 2: Load tokenizer using our function
+        model_name = test_repo_id.split("/")[-1]
+        tokenizer_dir = "tokenizer" if model_name == "FLUX.1-dev" else "."
+        tokenizer_path = os.path.join(self.temp_dir, model_name, tokenizer_dir)
+        our_tokenizer = build_hf_tokenizer(tokenizer_path)
+
+        # Step 3: Load tokenizer using official Tokenizer library (if available)
+        official_tokenizer = None
+        try:
+            official_tokenizer = Tokenizer.from_pretrained(test_repo_id)
+        except Exception as e:
+            print(f"Warning: Could not load official tokenizer for {test_repo_id}: {e}")
+
+        # Step 4: Load tokenizer using transformers AutoTokenizer (if available)
+        transformers_tokenizer = None
+        try:
+            from transformers import AutoTokenizer
+
+            transformers_tokenizer = AutoTokenizer.from_pretrained(test_repo_id)
```

Comment:
Will this test still needs to depend on `transformers` package? Do we need to add  `transformers` as a dependency in `requirement-dev.txt` to make this test work in CI?

Reply:
That's a good idea, I tested it locally since I have it installed but I think it makes sense to have it in dev as well.